### ISSUE TITLE
Using `argoWorkflow` causes deserialization error: unknown field "synchronization".

### ISSUE THREAD

**Issue by hisener** [argoproj/argo-events#800]
**Describe the bug**
Using Workflow Trigger ([`argoWorkflow`](https://github.com/argoproj/argo-events/blob/master/api/sensor.md#argoproj.io/v1alpha1.ArgoWorkflowTrigger)) causes deserialization error: unknown field "synchronization".

**To Reproduce**
1. Upgrade argo to `v2.10.0+`. If you are using argo-helm, it can be done by setting the image tag:

```
argo:
  images:
    tag: v2.10.0-rc4
```

2. Create a sensor with the following trigger template.

```
argoWorkflow:
  operation: submit
  source:
    resource:
      apiVersion: argoproj.io/v1alpha1
      kind: Workflow
      metadata:
        generateName: workflow-test-
      spec:
        entrypoint: step-1
        synchronization:
          semaphore:
            configMapKeyRef:
              name: synchronization-test
              key: workflow
```

If you check the created workflow, you should see the error message above.

**Expected behavior**
`synchronization` should be passed to the workflow trigger correctly.

**Workaround**
As a workaround, [K8STrigger](https://github.com/argoproj/argo-events/blob/master/api/sensor.md#argoproj.io/v1alpha1.StandardK8STrigger) can be used.
```
k8s:
  group: argoproj.io
  version: v1alpha1
  resource: workflows
  operation: create
  source:
    resource:
      apiVersion: argoproj.io/v1alpha1
      kind: Workflow
      metadata:
        generateName: workflow-test-
      spec:
        entrypoint: step-1
        synchronization:
          semaphore:
            configMapKeyRef:
              name: synchronization-test
              key: workflow
```

**Screenshots**
N/A

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
@hisener - What was the argo-events version?

**Comment by hisener** [comment]
v0.17.0

**Comment by whynowy** [comment]
@hisener - is that a new feature of `argo` 2.10.x? `argo-events` has the dependency on `argo` cli version, current version used is `2.9.x`. We can upgrade to `2.10.x` after it has an official release.

**Comment by hisener** [comment]
Yes, it's available in v2.10+. 👌
